Unstable test TestIssue20779

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
Found in CI https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_unit_test_ddlv1/detail/pull_unit_test_ddlv1/2679/pipeline/



### 2. What did you expect to see? (Required)
Tests pass stably

### 3. What did you see instead (Required)
```
[2025/05/27 08:53:25.791 +00:00] [INFO] [session.go:4148] ["CRUCIAL OPERATION"] [conn=1] [schemaVersion=56] [cur_db=test] [sql="create table t1(a int, b int, index idx(b));"] [user=]
    join_test.go:749: 
        	Error Trace:	pkg/executor/test/jointest/join_test.go:749
        	Error:      	An error is expected but got nil.
        	Test:       	TestIssue20779
[2025/05/27 08:53:25.836 +00:00] [ERROR] [misc.go:95] ["panic in the recoverable goroutine"] [r=testIssue20779] ["stack trace"="github.com/pingcap/tidb/pkg/util.WithRecovery.func1\n\tpkg/util/misc.go:97\nruntime.gopanic\n\tGOROOT/src/runtime/panic.go:791\ngithub.com/pingcap/tidb/pkg/executor/join.(*indexHashJoinOuterWorker).run\n\tpkg/executor/join/index_lookup_hash_join.go:398\ngithub.com/pingcap/tidb/pkg/executor/join.(*IndexNestedLoopHashJoin).startWorkers.func1\n\tpkg/executor/join/index_lookup_hash_join.go:173\ngithub.com/pingcap/tidb/pkg/util.WithRecovery\n\tpkg/util/misc.go:100"] [stack="github.com/pingcap/tidb/pkg/util.WithRecovery.func1\n\tpkg/util/misc.go:95\nruntime.gopanic\n\tGOROOT/src/runtime/panic.go:791\ngithub.com/pingcap/tidb/pkg/executor/join.(*indexHashJoinOuterWorker).run\n\tpkg/executor/join/index_lookup_hash_join.go:398\ngithub.com/pingcap/tidb/pkg/executor/join.(*IndexNestedLoopHashJoin).startWorkers.func1\n\tpkg/executor/join/index_lookup_hash_join.go:173\ngithub.com/pingcap/tidb/pkg/util.WithRecovery\n\tpkg/util/misc.go:100"]
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/executor/test/jointest/join_test.go
+++ b/pkg/executor/test/jointest/join_test.go
@@ -743,7 +743,7 @@ func TestIssue20779(t *testing.T) {
 		require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/pkg/executor/join/testIssue20779"))
 	}()
 
-	rs, err := tk.Exec("select /*+ inl_hash_join(t2) */ t1.b from t1 left join t1 t2 on t1.b=t2.b order by t1.b;")
+	rs, err := tk.Exec("select /*+ inl_hash_join(t2) */ t1.b from t1 use index(idx) left join t1 t2 use index(idx) on t1.b=t2.b order by t1.b;")
 	require.NoError(t, err)
 	_, err = session.GetRows4Test(context.Background(), nil, rs)
 	require.EqualError(t, err, "testIssue20779")

